src/nupic/algorithms/Connections.hpp

```diff
@@ -375,21 +376,12 @@ class Connections : public Serializable
    * @param activePresynapticCells
    * Active cells in the input.
    */
-  void
-  computeActivity(std::vector<SynapseIdx> &numActiveConnectedSynapsesForSegment,
-                  std::vector<SynapseIdx> &numActivePotentialSynapsesForSegment,
-                  const std::vector< CellIdx> &activePresynapticCells,
-                  const Permanence connectedPermanence) const;
+  void computeActivity(std::vector<SynapseIdx> &numActiveConnectedSynapsesForSegment,
+                       std::vector<SynapseIdx> &numActivePotentialSynapsesForSegment,
+                       const std::vector<CellIdx> &activePresynapticCells) const;
 
-  void
-  computeActivity(std::vector<SynapseIdx> &numActiveConnectedSynapsesForSegment,
-                  const std::vector<CellIdx> &activePresynapticCells) const;
-
-  void
-  computeActivity(std::vector<SynapseIdx> &numActiveConnectedSynapsesForSegment,
-                  std::vector<SynapseIdx> &numActivePotentialSynapsesForSegment,
-                  const CellIdx activePresynapticCell,
-                  const Permanence connectedPermanence) const;
+  void computeActivity(std::vector<SynapseIdx> &numActiveConnectedSynapsesForSegment,  //TODO remove the 2 arg version, use only the 3 arg (with 1 being optional)
```

Comment:
No can do, the two different versions do different things.  One computes less than have as much as the other, making it significantly faster.  Removing it would likely double the spatial poolers runtime.

Reply:
> One computes less than have as much as the other, making it significantly faster. 

I'm aware, if we use {} the loop will skip, or some empheral value, as in TM(?)

Reply:
#371